### PR TITLE
Intro to STMO document.

### PR DESCRIPTION
**Pull request by rafrombrc**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/firefox-data-docs/29)

### CODE REVIEW

File: tools/stmo.md [hunk 1]
```diff
@@ -1,5 +1,276 @@
...
+SQL provides the ability to extract and manipulate the data, but you won't get
+very far without having some familiarity with what data is actually available,
+and how that data is structured. Each query in STMO is associated with exactly
+one data source, and you have to know ahead of time which data source contains
+the information that you need. One of the most commonly used data sources is
+called *Athena* (referring to Amazon's [Athena](https://aws.amazon.com/athena/)
+query service, on which it is built), which contains most of the data that is
+obtained from telemetry pings received from Firefox clients. The *Athena*
+source is slowly replacing the *Presto* data source. *Presto* contains all of
+the data that's exposed via *Athena* and more, but returns query results much
```
Comment (harterrt) [inline tools/stmo.md]: Looks like coverage is spotty for now. Maybe we should file a bug against robotblake to have a list of what is and isn't supported?

File: tools/stmo.md [hunk 2]
```diff
@@ -1,5 +1,276 @@
...
+#### Creating A Query
+
+First we must create a query, which can be initiated by clicking on the 'New
+Query' button near the top left of the page. This will bring you to the query
+editing page:
+
+![New Query Page](/../assets/STMO_screenshots/new_query.png "New Query page")
+
+The first query will be counting the number of client ids that we have coming
+from each country, for the top N countries. There is a "client_count" data set
```
Comment (harterrt) [inline tools/stmo.md]: It feels like there should be a paragraph break between these sentences

Comment (harterrt) [inline tools/stmo.md]: Maybe link to the client_count dataset reference article:
http://docs.telemetry.mozilla.org/datasets/batch_view/client_count/reference.html

File: tools/stmo.md [hunk 3]
```diff
@@ -1,5 +1,276 @@
...
+that is generated from Firefox telemetry pings, so first we check to see if
+that data is available in the *Athena* data source by typing "client_count"
```
Comment (harterrt) [inline tools/stmo.md]: It may be helpful to break these steps into bullet points. It would be useful to be able to visually identify the splits between the tasks, e.g.:
* Check if the datasets in Athena
* Verify it's in Presto
* Take a look at the available columns

File: tools/stmo.md [hunk 4]
```diff
@@ -1,5 +1,276 @@
...
+column for all of its counting functionality. Converting the *hll* value back
+to a regular numeric value requires the use of the following magic SQL
+incantation:
+
+  ```cardinality(merge(cast(hll as HLL)))```
+
+So a query that extracts all of the unique country values and the count for each
+one, sorted from highest count to lowest count looks like this:
+
+  ```SELECT country, cardinality(merge(cast(hll AS HLL))) AS client_count FROM client_count GROUP BY 1 ORDER BY 2 DESC```
```
Comment (harterrt) [inline tools/stmo.md]: If you format this as follows you'll get syntax highlighting and you can use some whitespace:

```
```sql
SELECT 
  country,
  cardinality(merge(cast(hll AS HLL))) AS client_count
FROM client_count 
GROUP BY 1 
ORDER BY 2 DESC
```.

File: tools/stmo.md (hunk 1 above)
Comment (rafrombrc) [inline tools/stmo.md]: Is what true? I'm trying to say that Presto has everything but is slower, Athena doesn't have everything but is faster. Is that not what you're experiencing? Or maybe my wording is unclear here?

File: tools/stmo.md [hunk 5]
```diff
@@ -1,5 +1,276 @@
...
+called *Athena* (referring to Amazon's [Athena](https://aws.amazon.com/athena/)
+query service, on which it is built), which contains most of the data that is
+obtained from telemetry pings received from Firefox clients. The *Athena*
+source is slowly replacing the *Presto* data source. *Presto* contains all of
+the data that's exposed via *Athena* and more, but returns query results much
+more slowly.
+
+Other available data sources include *Crash-DB*, *Tiles*, *Sync Stats*, *Push*,
+*Test Pilot*, *ATMO*, and even a *re:dash metadata* which connects to STMO's
+own redash database. It is beyond the scope of this document to list and
```
Comment (rafrombrc) [inline tools/stmo.md]: I'm confused... the previous paragraph talks about the "Presto" and "Athena" data sets, which IIUC are the ones that contain the derived data sets. Is that not what you mean?

+1 to the link, I'll add that now.

File: tools/stmo.md (hunk 1 above)
Comment (harterrt) [inline tools/stmo.md]: Ach, sorry for the unclear comment. I misread this to mean Athena contains all of Presto's data (the opposite is true).

File: tools/stmo.md (hunk 5 above)
Comment (harterrt) [inline tools/stmo.md]: I think the link will take care of this comment. I was looking for some documentation on what datasets were available in athena, but this is out of scope for this document.

File: tools/stmo.md [hunk 6]
```diff
@@ -1,5 +1,276 @@
...
+
+#### A Note About Limits
+
+Once you save the visualization settings and return to the query source page,
+you should have a nice bar graph near the bottom of the page. You may notice,
+however, that the graph has quite a long tail. Rather than seeing *all* of
+the countries, it might be nicer to only see the top 20. We can do this by adding
+a `LIMIT` clause to the end of our query:
+
+  ```SELECT country, cardinality(merge(cast(hll AS HLL))) AS client_count FROM client_count GROUP BY 1 ORDER BY 2 DESC LIMIT 20```
```
Comment (harterrt) [inline tools/stmo.md]: As noted above, this query would be easier to read with some whitespace.

File: tools/stmo.md [hunk 7]
```diff
@@ -1,5 +1,276 @@
...
+you should get a new bar graph that only shows the 20 countries with the
+highest number of unique clients. In this case, the full data set has
+approximately 250 return values, and so limiting the result count improves
+readability. In other cases, however, an unlimited query might return thousands
+or even millions of rows. When those queries are run, readability is not the
+only problem; queries that return millions of rows can tax the system, failing
+to return the desired results, and negatively impacting the performance of all
+of the other users of the system. Thus, a very important warning:
+
+  ALL QUERIES SHOULD INCLUDE A "LIMIT" STATEMENT BY DEFAULT!
```
Comment (harterrt) [inline tools/stmo.md]: I had a hard time spotting this when skimming, can you bold as well?

**Comment by rafrombrc** [comment]
Okay @harterrt I've finally got around to integrating your suggestions. I've made tweaks according to your comments, made one additional formatting tweak, and added a link to more comprehensive redash docs at the end. I think we're ready for business, hopefully you do too.

**Comment by harterrt** [comment]
:lgtm:! Thanks again, Rob.

---

Review status: 0 of 8 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/firefox-data-docs/29#-:-Kpu86kKcP_JvTE4zDlO:b3c77jq)*

**Comment by harterrt** [comment]
Reviewed 7 of 8 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/firefox-data-docs/29)*
